File: 01gpio/03digital_in/ldr/ldr.js
var onoff = require('onoff');
var pin = new onoff.Gpio(18, 'in', 'both');

console.log(pin.readSync() == 0 ? 'dark': 'light');

pin.watch(function(err, value){
	console.log(value==0 ? 'dark' : 'light');
});

File: 01gpio/03digital_in/ldr/ldr.py
import RPi.GPIO as GPIO

PIN = 18
GPIO.setmode(GPIO.BCM)
GPIO.setup(PIN, GPIO.IN)
val = -1
try:
    while True:
        read = GPIO.input(PIN)
        if read != val:
            val = read
            print('dark' if val == 0 else 'light')
finally:
    print("clean up.")
    GPIO.cleanup()
